### PR TITLE
2670fix

### LINKED ISSUE #2670: entityNearest() on Plots.Bar doesn't return bars that extend off the bottom of the plot
Repro: http://jsfiddle.net/94g63w6g/

**Actual**: bar extending off the bottom of the `Plot` is never highlighted
**Expected**: hovering close to a bar should select it

Note that this doesn't happen for `entitiesAt()`: http://jsfiddle.net/a1ocp88b/

### PR DESCRIPTION
**Pull request by crmorford**
fixes #2670 

Category: 
Vertical Bars http://jsfiddle.net/bza0tsn9/1/
Horizontal Bars http://jsfiddle.net/bza0tsn9/2/

Numeric:
Vertical Bars http://jsfiddle.net/bza0tsn9/5/
Horizontal Bars http://jsfiddle.net/bza0tsn9/3/

### CODE REVIEW

File: src/plots/barPlot.ts [hunk 1]
```diff
@@ -327,14 +327,12 @@ export module Plots {
...
 
       let barBBox = {
-        x: pixelPoint.x - width / 2,
-        y: pixelPoint.y,
-        width: width,
-        height: height
+        x: attrToProjector["x"](datum, index, dataset),
+        y: attrToProjector["y"](datum, index, dataset),
+        width: attrToProjector["width"](datum, index, dataset),
+        height: attrToProjector["height"](datum, index, dataset)
```
Comment (bluong) [inline src/plots/barPlot.ts]: Not only is this more correct, but it also makes a heck lot more sense...

File: test/plots/barPlotTests.ts [hunk 2]
```diff
@@ -32,6 +32,96 @@ describe("Plots", () => {
...
+      ];
+      let dataset = new Plottable.Dataset(data);
+      let barPlot = new Plottable.Plots.Bar<string, number>();
+      barPlot.addDataset(dataset);
+      yScale.domain([-1, 1]);
+      barPlot.x((d) => d.x, xScale);
+      barPlot.y((d) => d.y, yScale);
+      barPlot.renderTo(svg);
+
+      let positiveBar =  barPlot.entities()[0];
```
Comment (bluong) [inline test/plots/barPlotTests.ts]: Double whitespace here for some reason

File: test/plots/barPlotTests.ts [hunk 3]
```diff
@@ -32,6 +32,96 @@ describe("Plots", () => {
...
+      ];
+      let dataset = new Plottable.Dataset(data);
+      let barPlot = new Plottable.Plots.Bar<number, string>("horizontal");
+      barPlot.addDataset(dataset);
+      xScale.domain([-1, 1]);
+      barPlot.x((d) => d.x, xScale);
+      barPlot.y((d) => d.y, yScale);
+      barPlot.renderTo(svg);
+
+      let positiveBar =  barPlot.entities()[0];
```
Comment (bluong) [inline test/plots/barPlotTests.ts]: Double whitespace here...

**Comment by bluong** [comment]
Just the whitespace issues.

**Comment by jtlan** [comment]
Looks good!

**Pull request merged**